internal/cli/cmd/auth/authorize/cached_provider.go

```diff
@@ -0,0 +1,315 @@
+	}
+}
+
+func defaultKeyring() (keyring.Keyring, bool) {
+	k, err := keyring.Open(keyring.Config{
+		AllowedBackends: []keyring.BackendType{
+			keyring.SecretServiceBackend,
+			keyring.KWalletBackend,
+			keyring.KeychainBackend,
+			keyring.WinCredBackend,
+		},
+		ServiceName:              keyringService,
+		KeychainTrustApplication: true,
+		KeychainSynchronizable:   true,
+	})
+
+	if err != nil {
+		return nil, false
+	}
+	return k, true
+}
+
+func (k *KeyringCachedTokenProvider) GetTokens() (*authenticator.TokenResponse, error) {
+	if !k.keyringCached.isValid() {
+		token, tokenErr := k.fileCached.readToken()
+		if os.IsNotExist(tokenErr) {
+			return nil, nil
+		}
+		return token, tokenErr
+	}
+
+	data, err := k.keyringCached.get()
+	if err != nil {
+		if err == keyring.ErrKeyNotFound {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "error getting token information from keyring")
+	}
+
+	var tokenResponse authenticator.TokenResponse
+	err = json.Unmarshal(data, &tokenResponse)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not unmarshal token data from keyring")
+	}
+
+	return &tokenResponse, nil
+}
+
+func (k *KeyringCachedTokenProvider) cacheTokens(tokenResponse *authenticator.TokenResponse) error {
+	data, err := json.Marshal(tokenResponse)
+	if err != nil {
+		return errors.Wrap(err, "could not marshal token data for keyring")
+	}
+
+	if !k.keyringCached.isValid() {
+		return k.fileCached.writeToken(data)
+	}
+
+	return k.keyringCached.set(data)
+}
+
+func (k *KeyringCachedTokenProvider) deleteTokens() error {
+	if !k.keyringCached.isValid() {
+		return k.fileCached.deleteToken()
+	}
+
+	return k.keyringCached.remove()
+}
+
+func (k *KeyringCachedTokenProvider) cacheUserInfo(userInfo *authenticator.UserInfoResponse) error {
+	saveDir, err := k.fileCached.getConfigDir()
+	if err != nil {
+		return err
+	}
+
+	if err := os.MkdirAll(saveDir, os.ModePerm); err != nil {
+		return errors.Wrap(err, "failed to create config directory")
+	}
+	savePath := filepath.Join(saveDir, k.fileCached.userInfoFilename)
+
+	newData, err := json.Marshal(userInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal user info")
+	}
+
+	if err := os.WriteFile(savePath, newData, fileMode); err != nil {
+		return errors.Wrap(err, "failed to write user info file")
+	}
+	return nil
+}
+
+func (k *KeyringCachedTokenProvider) getUserInfo() (*authenticator.UserInfoResponse, error) {
+	saveDir, err := k.fileCached.getConfigDir()
+	if err != nil {
+		return nil, err
+	}
+	savePath := filepath.Join(saveDir, k.fileCached.userInfoFilename)
+	data, err := os.ReadFile(savePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read user info file")
+	}
+
+	var userInfo authenticator.UserInfoResponse
+	if err := json.Unmarshal(data, &userInfo); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal user info")
+	}
+	return &userInfo, nil
+}
+
+func (f *FileCached) getConfigDir() (string, error) {
```

Comment:
Use a subdirectory such as `~/.kbcli/auth` to store authorize config.  https://github.com/apecloud/kubeblocks/pull/4123/files#diff-9f2eb3f0bfa1a95f11d959906326c2271e46f2f56dbe1059508489137d7b27f1R308 will remove this config dir, it is not expected to remove `~/.kbcli`

Reply:
ACK